src/wutil.cpp

```diff
@@ -203,6 +203,27 @@ int make_fd_blocking(int fd) {
     return err == -1 ? errno : 0;
 }
 
+maybe_t<wcstring> wreadlink(const wcstring &file_name) {
+    struct stat buf;
+    if (lwstat(file_name, &buf) == -1) {
+        return none();
+    }
+    ssize_t bufsize = buf.st_size + 1;
+    char target_buf[bufsize];
+    const std::string tmp = wcs2string(file_name);
```

Comment:
(note: We really really do way too much of this wcs2string and back - this will do wcs2string for the `lwstat` above, and then wcs2string again for the readlink. In some benchmarks ~50% of the time is taken in widening and narrowing strings)

(but since this is just for the error, it's okay for now - we should just probably start doing all the path stuff in narrow exclusively, so you can keep one narrow string around if you need to do more than one thing)